UserService/app/src/main/java/com/bookmyshow/userservice/controller/PermissionController.java

```diff
@@ -0,0 +1,48 @@
+package com.bookmyshow.userservice.controller;
+
+import com.bookmyshow.userservice.dto.PermissionRequestDTO;
+import com.bookmyshow.userservice.dto.PermissionResponseDTO;
+import com.bookmyshow.userservice.service.PermissionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/api/v1")
+public class PermissionController {
+
+    @Autowired
+    private PermissionService service;
+
+    @PostMapping(path = "/permissions")
+    public ResponseEntity<PermissionResponseDTO> addPermission(@Valid @RequestBody PermissionRequestDTO permissionRequestDTO){
+        return new ResponseEntity( service.addPermission(permissionRequestDTO) , HttpStatus.CREATED);
+    }
+
+    @GetMapping(path = "/permissions")
```

Comment:
may be we can create an AbstractController and extend all these controllers so that service can be injected thru runtime polymorphism.